util/uring/submit_entry.h

```diff
@@ -1,214 +1,3 @@
-
-  void PrepWriteFixed(int fd, void* buf, unsigned size, size_t offset, uint16_t buf_index) {
-    PrepFd(IORING_OP_WRITE_FIXED, fd);
-    sqe_->addr = (unsigned long)buf;
-    sqe_->len = size;
-    sqe_->off = offset;
-    sqe_->buf_index = buf_index;
-  }
-
-  void PrepWriteV(int fd, const struct iovec* vec, unsigned nr_vecs, size_t offset,
-                  unsigned flags = 0) {
-    PrepFd(IORING_OP_WRITEV, fd);
-    sqe_->addr = (unsigned long)vec;
-    sqe_->len = nr_vecs;
-    sqe_->off = offset;
-    sqe_->rw_flags = flags;
-  }
-
-  void PrepFallocate(int fd, int mode, off_t offset, off_t len) {
-    PrepFd(IORING_OP_FALLOCATE, fd);
-    sqe_->off = offset;
-    sqe_->len = mode;
-    sqe_->addr = uint64_t(len);
-  }
-
-  void PrepFadvise(int fd, off_t offset, off_t len, int advice) {
-    PrepFd(IORING_OP_FADVISE, fd);
-    sqe_->fadvise_advice = advice;
-    sqe_->len = len;
-    sqe_->off = offset;
-  }
-
-  void PrepSend(int fd, const void* buf, size_t len, unsigned flags) {
-    PrepFd(IORING_OP_SEND, fd);
-    sqe_->addr = (unsigned long)buf;
-    sqe_->len = len;
-    sqe_->msg_flags = flags;
-  }
-
-  void PrepSendMsg(int fd, const struct msghdr* msg, unsigned flags) {
-    PrepFd(IORING_OP_SENDMSG, fd);
-    sqe_->addr = (unsigned long)msg;
-    sqe_->len = 1;
-    sqe_->msg_flags = flags;
-  }
-
-  void PrepConnect(int fd, const struct sockaddr* addr, socklen_t addrlen) {
-    PrepFd(IORING_OP_CONNECT, fd);
-    sqe_->addr = (unsigned long)addr;
-    sqe_->len = 0;
-    sqe_->off = addrlen;
-  }
-
-  void PrepClose(int fd) {
-    PrepFd(IORING_OP_CLOSE, fd);
-  }
-
-  void PrepTimeout(const timespec* ts, bool is_abs = true) {
-    PrepFd(IORING_OP_TIMEOUT, -1);
-    sqe_->addr = (unsigned long)ts;
-    sqe_->len = 1;
-    sqe_->timeout_flags = (is_abs ? IORING_TIMEOUT_ABS : 0);
-  }
-
-  void PrepTimeoutRemove(unsigned long long userdata) {
-    PrepFd(IORING_OP_TIMEOUT_REMOVE, -1);
-    sqe_->addr = userdata;
-    sqe_->len = 1;
-  }
-
-  // Sets up link timeout with relative timespec.
-  void PrepLinkTimeout(const timespec* ts) {
-    PrepFd(IORING_OP_LINK_TIMEOUT, -1);
-    sqe_->addr = (unsigned long)ts;
-    sqe_->len = 1;
-    sqe_->timeout_flags = 0;
-  }
-
-  // how is either: SHUT_RD, SHUT_WR or SHUT_RDWR.
-  void PrepShutdown(int fd, int how) {
-    PrepFd(IORING_OP_SHUTDOWN, fd);
-    sqe_->len = how;
-  }
-
-  void PrepMsgRing(int fd, unsigned int len, uint64_t data) {
-    PrepFd(IORING_OP_MSG_RING, fd);
-    sqe_->len = len;
-    sqe_->off = data;
-    sqe_->rw_flags = 0;
-  }
-
-  io_uring_sqe* sqe() {
-    return sqe_;
-  }
-
-  void PrepNOP() {
-    PrepFd(IORING_OP_NOP, -1);
-  }
-
-  // Used only by Proactor.
-  explicit SubmitEntry(io_uring_sqe* sqe) : sqe_(sqe) {
-  }
-
- private:
-  void PrepFd(int op, int fd) {
-    sqe_->opcode = op;
-    sqe_->fd = fd;
-  }
-
-#ifndef USE_FB2
-  friend class Proactor;
-#endif
-};
-
-}  // namespace uring
-}  // namespace util
+#include "util/fibers/submit_entry.h"
```

Comment:
are those headers deprecated now? Can you document it (or put a `#pragma message("deprecated")`)?

Reply:
fixed